File: src/main/java/ac/dnd/server/admission/infrastructure/persistence/converter/StringHmacConverter.java
package ac.dnd.server.admission.infrastructure.persistence.converter;

import org.springframework.stereotype.Component;

import ac.dnd.server.admission.infrastructure.persistence.crypto.HmacBlindIndexCreator;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter
@Component
public class StringHmacConverter implements AttributeConverter<String, String> {

	private final HmacBlindIndexCreator hmacBlindIndexCreator;

	public StringHmacConverter(final HmacBlindIndexCreator hmacBlindIndexCreator) {
		this.hmacBlindIndexCreator = hmacBlindIndexCreator;
	}

	@Override
	public String convertToDatabaseColumn(final String attribute) {
		return hmacBlindIndexCreator.create(attribute);
	}

	@Override
	public String convertToEntityAttribute(final String dbData) {
		return dbData;
	}

}

File: src/main/kotlin/ac/dnd/server/admission/infrastructure/persistence/converter/StringHmacConverter.kt
package ac.dnd.server.admission.infrastructure.persistence.converter

import ac.dnd.server.admission.infrastructure.persistence.crypto.HmacBlindIndexCreator
import org.springframework.stereotype.Component
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
@Component
class StringHmacConverter(
    private val hmacBlindIndexCreator: HmacBlindIndexCreator
) : AttributeConverter<String, String> {

    override fun convertToDatabaseColumn(attribute: String): String {
        return hmacBlindIndexCreator.create(attribute)
    }

    override fun convertToEntityAttribute(dbData: String): String {
        return dbData
    }
}